### ISSUE TITLE
Unnecessary storage writes in mint loops?

### ISSUE THREAD

**Issue by RogerPodacter** [FrankNFT-labs/ERC721F#81]
Apologies if I'm reading this incorrectly, but in code like this:

```solidity
unchecked {
    for (uint256 i; i < numberOfTokens; ) {
        _mint(msg.sender, supply + i); // no need to use safeMint as we don't allow contracts.
        i++;
    }
}
```

It looks like we're updating `_tokenSupply` every loop. In addition we are calling OZ's `_mint()` which will also do `_balances[to] += 1` on every loop. OZ also does two existence checks which you don't need because you have ascending ids.

Just a thought!

**Comment by FrankNFT-labs** [comment]
so, create a bulk mint method like the ERC721A has, and work around those extra writes would be a solution. 
Thx a lot for this suggestion, will defo look into it.

**Comment by FrankNFT-labs** [comment]
Closed in PR #90
